Compiler error: Interface failed to work with embedded struct

<!-- It is advisable to update all relevant modules using `v outdated` and `v install` -->
**V doctor:**
```
OS: windows, Microsoft Windows 10 Pro v19041 64-bit
Processor: 4 cpus, 64bit, little endian, AMD Athlon 3000G with Radeon Vega Graphics
CC version: cc (Rev5, Built by MSYS2 project) 10.3.0

getwd: D:\Projects\v-ground\interface
vmodules: C:\Users\User\.vmodules
vroot: D:\Tools\V
vexe: D:\Tools\V\v.exe
vexe mtime: 2021-12-21 09:54:42
is vroot writable: true
is vmodules writable: true
V full version: V 0.2.4 cc2fcb5

Git version: git version 2.32.0.windows.1
Git vroot status: weekly.2021.51-1-gcc2fcb52-dirty
.git/config present: true
thirdparty/tcc status: thirdparty-windows-amd64 9e2fc2ca
```

**What did you do?**
`v -g -o vdbg cmd/v && vdbg .\main.v`
```v
module main

// 
pub interface IObject {
	mut:
		do_stuff()
		do_something_with_int(int)
		do_something_with_string(string)
}

pub struct BaseObject {
	mut:
		some_attr int
}

pub fn (mut base BaseObject) do_stuff() {}
pub fn (mut base BaseObject) do_something_with_int(n int) {}
pub fn (mut base BaseObject) do_something_with_string(s string) {}
pub fn (mut base BaseObject) method_thats_available_to_all_object() {}


pub struct GameObject {
	BaseObject

	pub mut:
		some_attr_for_game int
}

//
fn main() {
	mut common_object := []IObject{}
	common_object << GameObject{} // Fails
}

```

**What did you expect to see?**
Expecting the code to compile.


**What did you see instead?**
```
==================
C:/Users/User/AppData/Local/Temp/v_0/main.6861163767165800170.tmp.c:453: warning: WINVER redefined
C:/Users/User/AppData/Local/Temp/v_0/main.6861163767165800170.tmp.c:1932: error: 'main__GameObject_do_stuff' undeclared
...
==================
(Use `v -cg` to print the entire error message)

builder error: 
==================
C error. This should never happen.

This is a compiler bug, please report it using `v bug file.v`.

https://github.com/vlang/v/issues/new/choose

You can also use #help on Discord: https://discord.gg/vlang```

--- a/vlib/v/gen/c/cgen.v
+++ b/vlib/v/gen/c/cgen.v
@@ -7383,11 +7383,12 @@ static inline $interface_name I_${cctype}_to_Interface_${interface_name}($cctype
 							false)
 					}
 				}
-				mut method_call := '${cctype}_$name'
+				styp := g.cc_type(method.params[0].typ, true)
+				mut method_call := '${styp}_$name'
 				if !method.params[0].typ.is_ptr() {
 					// inline void Cat_speak_Interface_Animal_method_wrapper(Cat c) { return Cat_speak(*c); }
 					iwpostfix := '_Interface_${interface_name}_method_wrapper'
-					methods_wrapper.write_string('static inline ${g.typ(method.return_type)} $method_call${iwpostfix}(')
+					methods_wrapper.write_string('static inline ${g.typ(method.return_type)} ${cctype}_$name${iwpostfix}(')
 					//
 					params_start_pos := g.out.len
 					mut params := method.params.clone()

--- a/vlib/v/tests/interface_method_using_struct_embed_test.v
+++ b/vlib/v/tests/interface_method_using_struct_embed_test.v
@@ -1,0 +1,34 @@
+module main
+
+pub interface IObject {
+mut:
+	do_stuff()
+	do_something_with_int(int)
+	do_something_with_string(string)
+}
+
+pub struct BaseObject {
+mut:
+	some_attr int
+}
+
+pub fn (mut base BaseObject) do_stuff() {}
+
+pub fn (mut base BaseObject) do_something_with_int(n int) {}
+
+pub fn (mut base BaseObject) do_something_with_string(s string) {}
+
+pub fn (mut base BaseObject) method_thats_available_to_all_object() {}
+
+pub struct GameObject {
+	BaseObject
+pub mut:
+	some_attr_for_game int
+}
+
+fn test_interface_method_using_struct_embed() {
+	mut common_object := []IObject{}
+	common_object << GameObject{}
+	println(common_object)
+	assert common_object.len == 1
+}

